aws/resource_aws_eks_node_group.go

```diff
@@ -384,25 +385,25 @@ func resourceAwsEksNodeGroupRead(d *schema.ResourceData, meta interface{}) error
 	d.Set("release_version", nodeGroup.ReleaseVersion)
 
 	if err := d.Set("remote_access", flattenEksRemoteAccessConfig(nodeGroup.RemoteAccess)); err != nil {
-		return fmt.Errorf("error setting remote_access: %w", err)
+		return diag.Errorf("error setting remote_access", err)
 	}
 
 	if err := d.Set("resources", flattenEksNodeGroupResources(nodeGroup.Resources)); err != nil {
-		return fmt.Errorf("error setting resources: %w", err)
+		return diag.Errorf("error setting resources", err)
 	}
 
 	if err := d.Set("scaling_config", flattenEksNodeGroupScalingConfig(nodeGroup.ScalingConfig)); err != nil {
-		return fmt.Errorf("error setting scaling_config: %w", err)
+		return diag.Errorf("error setting scaling_config", err)
 	}
 
 	d.Set("status", nodeGroup.Status)
 
 	if err := d.Set("subnet_ids", aws.StringValueSlice(nodeGroup.Subnets)); err != nil {
-		return fmt.Errorf("error setting subnets: %w", err)
+		return diag.Errorf("error setting subnets", err)
```

Comment:
Still need to pass `error`, just can't use `%w`.

```suggestion
		return diag.Errorf("error setting subnets: %s", err)
```